streams/src/test/java/org/apache/kafka/streams/tests/SmokeTestDriver.java

```diff
@@ -106,6 +108,7 @@ int next() {
     public static Map<String, Set<Integer>> generate(final String kafka,
                                                      final int numKeys,
                                                      final int maxRecordsPerKey,
+                                                     final Duration timeToSpend,
```

Comment:
good call!

Reply:
hey @vvcephei I'm wondering if we can simplify the code further: with `timeToSpend` now can we just remove autoTerminate?

1) SmokeTestDriverIntegrationTest.java: we are sending 1000 * 10 = 10K records only, so likely that will stop even before the second instance can be started, similar to the issue you observed in system test. I'd suggest we just set `timeToSpend` to 10 seconds so we have some enough time to start up to 10 / 1 (sleep time) = 10 instances.

2) As for StreamsSmokeTest itself, we only disableAutoTerminate in three of StreamsUpgradeTest cases, since we do not need to verify the number of records sent / received but only check upgrade completed. We can also use 30 seconds (not sure if it is sufficient, but we can get on average how much time those three tests will take from our nightlies), and even if the test completes before that it will call `driver.stop` to force-stop the driver anyways.

Reply:
Of course we can then remove the whole `disableAutoTerminate` thing from StreamsSmokeTest.

Reply:
Hmm, this is an interesting idea. I'll look into it.

That's a very good point about the integration test, I'll look at that as well.

Reply:
Ok, After trying to simplify the code, I found that it was actually better to split the two use cases, so that's what I did. It lets us ditch a whole bunch of bookkeeping when we just want to generate until the process is killed, and it also lets us skip a bunch of mind-bending boolean logic when we do want to just generate a fixed amount of data.

I also added a duration of 20 seconds to the integration test, and it didn't increase the run-time of the test, but now we're getting the same guarantee that we get with the system tests now.